src/components/common/InfiniteWrapper.tsx

```diff
@@ -0,0 +1,62 @@
+import styled from '@emotion/styled';
+import type { ReactNode } from 'react';
+import React from 'react';
+import { useInView } from 'react-intersection-observer';
+import type { InfiniteData, QueryFunctionContext, QueryKey } from 'react-query';
+
+import loadingGif from '@/assets/loading.gif';
+import { useInfiniteData } from '@/utils/useInfiniteData';
+
+interface PageResponse {
+  nextPageToken: string | null;
+}
+
+interface InfiniteWrapperProps<TData extends PageResponse> {
+  queryKey: QueryKey;
+  queryFn: (context: QueryFunctionContext<QueryKey>) => Promise<TData>;
+  children: (data: InfiniteData<TData>) => ReactNode;
+}
+
+export const InfiniteWrapper = <TData extends PageResponse>({
```

Comment:
제너릭이 필요한 상황은 아닌 것 같아 제거해도 좋겠네요 ㅎㅎ

```suggestion
export const InfiniteWrapper = <TData extends PageResponse>({
```